src/azure-cli/azure/cli/command_modules/network/_params.py

```diff
@@ -1348,8 +1348,27 @@ def load_arguments(self, _):
         c.argument('filters', type=get_json_object)
 
     with self.argument_context('network watcher flow-log') as c:
+        c.argument('network_watcher_name', options_list='--watcher')
+        c.ignore('location')
+        c.argument('flow_log_name', name_arg_type, help='The name of the flow logger', min_api='2019-11-01')
         c.argument('nsg', help='Name or ID of the network security group.')
-        c.argument('enabled', arg_type=get_three_state_flag())
+        c.argument('enabled', arg_type=get_three_state_flag(), help='Enable logging', default='true')
+        c.argument('retention', type=int, help='Number of days to retain logs')
+        c.argument('storage_account', help='Name or ID of the storage account in which to save the flow logs')
+        c.argument('storage_account_id', help='ID of the storage account in which to save the flow logs')
+
+    # temporary solution for compatible with old show command's parameter
+    # after old show command's parameter is deprecated and removed,
+    # this argument group "network watcher flow-log show" should be removed
+    with self.argument_context('network watcher flow-log show') as c:
+        c.argument('nsg',
+                   deprecate_info=c.deprecate(redirect='--watcher and --name combination', hide=False),
+                   help='Name or ID of the network security group.')
+
+    with self.argument_context('network watcher flow-log show',
+                               arg_group='Result with Azure Management Resource (ARM) Formatted') as c:
+        c.argument('network_watcher_name', options_list='--watcher')
```

Comment:
I think we can remove the above `c.argument`s. Right? Duplicate code may cause confusion, especially we have such a overwriting framework.

Reply:
Removed.